presto-common/src/main/java/com/facebook/presto/common/type/TypeUtils.java

```diff
@@ -101,4 +107,23 @@ static void checkElementNotNull(boolean isNull, String errorMsg)
             throw new NotSupportedException(errorMsg);
         }
     }
+
+    static <V> void validateEnumEntries(String enumName, Map<String, V> entries)
+    {
+        requireNonNull(enumName, "Enum name cannot be null");
+        requireNonNull(entries, "Enum entries cannot be null");
+        if (entries.containsKey(null)) {
+            throw new NotSupportedException(String.format("Enum %s cannot contain null key", enumName));
+        }
+        Set<V> visited = new HashSet<>();
+        for (V value : entries.values()) {
```

Comment:
This can be simplified as
```
checkArgument(entries.values().stream().filter(value -> value != null).collect(toImmutableSet()).size() == entries.size(), format("Enum %s cannot contain null or duplicate values", enumName));
```

You can use checkArgument for null check on key as well. These should be `IllegalArgumentException`.